### ISSUE TITLE
Secondary Color (Settings)

### ISSUE THREAD

**Issue by jneuendorf** [atom-material/atom-material-ui#410]
I first thought there was a bug concerning the persistency of the secondary color's value.
This was due to the fact that toggling `Generate complementary secondary color` did not change the secondary color value on the fly.

**Reproduction Steps:**

1. Open `atom-material-ui`'s settings from Atom -> Settings -> Themes
2. Uncheck `Generate complementary secondary color`
3. Change the secondary color.
4. Check `Generate complementary secondary color`
5. => no change

**Expected behavior:**

When toggling `Generate complementary secondary color` the value of the secondary color should change immediately.
Thus the colored square should change its color and the applied editor styles should also change.

**Observed behavior:**

Checking `Generate complementary secondary color` after choosing a secondary color (while `Generate complementary secondary color` was unchecked) has no effect (only after reloading the window).

**Atom version:** 1.17.0
**OS and version:** macOS Sierra 10.12.4 (16E195)

**Installed packages:**

```
Community Packages (45) ~/.atom/packages
├── MagicPython@1.0.10
├── Sublime-Style-Column-Selection@1.7.4
├── atom-beautify@0.29.24
├── atom-material-syntax@1.0.6
├── atom-material-ui@2.0.3
├── autocomplete-paths@1.0.5
├── autocomplete-php@0.3.7
├── autocomplete-ruby@0.2.8
├── autocomplete-sass@0.1.0
├── block-comment@0.4.3
├── busy-signal@1.4.1
├── code-annotations@2.1.3
├── color-picker@2.2.5
├── dash@1.7.1
├── django-templates@0.5.0
├── emmet@2.4.3
├── export-html@3.1.0
├── file-icons@2.1.6
├── highlight-selected@0.13.1
├── html-entities@0.4.0
├── hyperclick@0.0.40
├── imdone-atom@2.2.6
├── intentions@1.1.2
├── js-hyperclick@1.11.0
├── language-latex@1.0.0
├── language-spacebars@0.2.2
├── linter@2.1.4
├── linter-coffeelint@1.2.1
├── linter-flake8@2.2.1
├── linter-htmlhint@1.3.3
├── linter-jshint@3.1.2
├── linter-php@1.3.2
├── linter-ruby@1.2.5
├── linter-ui-default@1.5.4
├── merge-conflicts@1.4.5
├── minimap@4.28.2
├── minimap-git-diff@4.3.1
├── minimap-highlight-selected@4.6.1
├── minimap-pigments@0.2.2
├── pdf-view@0.58.0
├── pigments@0.39.1
├── project-manager@3.3.4
├── sort-lines@0.14.0
├── svg-preview@0.11.0
└── tabs-to-spaces@1.0.3
```

**Comment by wyqydsyq** [comment]
It appears that this feature is completely broken, having `Generate complementary secondary color` enabled causes the secondary colour to reset to `#FF0000` upon restarting Atom.

**Comment by silvestreh** [comment]
The feature *is* working as intended. This probably just needs a different label. After enabling the feature you should not get a new color, it only happens after changing the primary color or, given how the `atom.config.observe` works, when the theme is reloaded you should also get a new secondary color.

@wyqydsyq I cannot reproduce that. If your primary color is `#00CC00` you'd get `#FF0000`. Would you mind filing a new issue with more info?

**Comment by wyqydsyq** [comment]
Oh I guess that's working as intended then. Perhaps the option should be renamed "Generate contrasting secondary color" because `#FF0000` contrasts with `#00FFFF` more than compliments it.

**Comment by silvestreh** [comment]
They _are_ complementary colors in the RGB color model 😄 

![image](https://cloud.githubusercontent.com/assets/1133339/26333986/3e5888bc-3f38-11e7-8a45-5a7230399a88.png)
